runtime/altair/src/xcm.rs

```diff
@@ -87,6 +87,9 @@ impl frame_support::traits::Contains<RuntimeCall> for SafeCallFilter {
 			) | RuntimeCall::XcmpQueue(..)
 				| RuntimeCall::DmpQueue(..)
 				| RuntimeCall::Proxy(..)
+				| RuntimeCall::LiquidityPoolsGateway(
+					pallet_liquidity_pools_gateway::Call::process_msg { .. }
```

Comment:
I am not sure what we are filtering here. All these calls will be allowed, right?

* We need to allow `pallet_liquidity_pools_gateway::Call::process_msg { .. }`
* But I would rather not allow all of the above calls. Only some.

Why do we have them here and why did we choose them? ^^

Reply:
These are the allowed calls, exactly. The above ones are mirroring the allowed ones on [Polkadot](https://github.com/paritytech/polkadot/blob/52209dcfe546ff39cc031b92d64e787e7e8264d4/runtime/polkadot/src/xcm_config.rs#L182) and [System parachains](https://github.com/paritytech/cumulus/blob/d4bb2215bb28ee05159c4c7df1b3435177b5bf4e/parachains/runtimes/assets/asset-hub-polkadot/src/xcm_config.rs#L198). We can definitely be more restrictive.

Reply:
ok. As we are not a system parachain and do not want to be "controlled" by the relay I think we should revisit that. But lets keep it as is for now.

Reply:
I fully agree. In the end, it is much more restrictive than the current setting of allowing everything. I am happy to make another proposal and cut even more. My approach was to make the most open proposal first and then adjust based on your input.

Reply:
I still would opt for changing this. I would say remove everything and only add
* calls of `pallet-investments`
* `batch` and `batch-all` of `pallet-utility`
* calls of `pallet-order-book`


Do see anything else that would be needed to generate an xcm message that would allow somebody to invest `natively` from other parachains?

Reply:
We must not add `batch` and `batch-all` per definition! The `SafeCallFilter` should only allow calls which have 
> /// 1. Have a fixed weight;
/// 2. Cannot lead to another call being made;
/// 3. Have a defined proof size weight, e.g. no unbounded vecs in call
/// parameters. 

Moreover, as `pallet-investments` is not benchmarked, we must not allow that either.

We can add `pallet-orderbook` calls though.

Reply:
Makes sense. Then lets just allow the one that are safe, add the order book. But we must add `pallet-investments` soon in order to allow somebody from asset hub to invest directly. 

I still would like to remove the system extrinsics though. Although, the origin is not correct. Not a blocker.

Reply:
[cea48d4](https://github.com/centrifuge/centrifuge-chain/pull/1526/commits/cea48d46ff5ee057f18c873591c60eb34a934357)

Reply:
Had to remove the `OrderBook` call again as its not implemented for Altair or Centrifuge Runtime.

Reply:
>We can do this ones we have reasonable upper PoV bounds, most ideally real benchmarks. Beforehand, we are vulnerable to PoV excess and thus stalling.

Yeah, good catch. Thanks for being so thorough!